src/mon/MonClient.cc

```diff
@@ -709,28 +715,30 @@ void MonClient::tick()
   } else if (!cur_mon.empty()) {
     // just renew as needed
     utime_t now = ceph_clock_now(cct);
-    ldout(cct, 10) << "renew subs? (now: " << now 
-		   << "; renew after: " << sub_renew_after << ") -- " 
-		   << (now > sub_renew_after ? "yes" : "no") 
-		   << dendl;
-    if (now > sub_renew_after)
-      _renew_subs();
+    if (!cur_con->has_feature(CEPH_FEATURE_MON_STATEFUL_SUB)) {
+      ldout(cct, 10) << "renew subs? (now: " << now
+		     << "; renew after: " << sub_renew_after << ") -- "
+		     << (now > sub_renew_after ? "yes" : "no")
+		     << dendl;
+      if (now > sub_renew_after)
+	_renew_subs();
+    }
 
     cur_con->send_keepalive();
 
     if (state == MC_STATE_HAVE_SESSION) {
-      send_log();
```

Comment:
Why don't you want to send the log if reconnecting? The information in it might still be useful.
(Commit "mon/MonClient: don't send log if we're reconnecting")